tools/autograd/gen_variable_type.py

```diff
@@ -689,7 +689,7 @@ def unpack_args(env, declaration):
 
             dynamic_type = arg['dynamic_type']
             is_nullable = arg.get('is_nullable', False)
-            ref = (not is_nullable) and dynamic_type != 'TensorList'
+            ref = (not is_nullable) and dynamic_type not in ['TensorList', 'THSTensor*']
```

Comment:
this looks wrong, because you are mixing TH and ATen types.  The dynamic_type of the argument should be "SparseTensor" in Declarations.yaml, matching e.g. IndexTensor:
```
 - dynamic_type: IndexTensor
    name: index
    type: const Tensor &
```

you probably have to add THSTensor to DYNAMIC_TYPE.

Reply:
Done